File: number58_2.cpp
#include <stdio.h>
using namespace std;
int DFS(int n,int x){
	if(x>n){
		return 0;
	}
	else{
		DFS(n,2*x);
		printf("%d",x);
		DFS(n,2*x+1);
	}
}
int main() { 
// 58. 이진트리 깊이우선트리(DFS)
// 중위순회 출력.
	int n;
	printf("노드의 갯수를 입력하세요:");
	scanf("%d",&n);
	DFS(n,1);
	return 0;
 
}

File: Number58_2.java
package algorithm;
import java.util.Scanner;
public class Number58_2 {

	public static void DFS(int n, int x) {
		if(x>n) return;
		else {
			DFS(n,x*2);
			System.out.print(x);
			DFS(n,x*2+1);
		}
	}

	public static void main(String[] args) {
		// 58_2. 이진트리 깊이우선트리(DFS)
		// 중위순회 출력
		System.out.print("노드의 갯수를 입력하세요:");
		Scanner stdIn = new Scanner(System.in);
		int n = stdIn.nextInt();
		int x = n;
		DFS(n,1);
	}

}
